action/protocol/staking/handlers.go

```diff
@@ -775,6 +776,86 @@ func (p *Protocol) handleCandidateUpdate(ctx context.Context, act *action.Candid
 	return log, nil
 }
 
+func (p *Protocol) handleProposerRegister(ctx context.Context, act *action.ProposerRegister, esm *executorStateManager,
+) (*receiptLog, []*action.TransactionLog, error) {
+	actCtx := protocol.MustGetActionCtx(ctx)
+	blkCtx := protocol.MustGetBlockCtx(ctx)
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+	log := newReceiptLog(p.addr.String(), HandleExecutorRegister, featureCtx.NewStakingReceiptFormat)
+
+	actAmount := act.Amount()
+	registrationFee := new(big.Int).Set(p.config.RegistrationConsts.Fee)
+
+	caller, fetchErr := fetchCaller(ctx, esm, new(big.Int).Add(actAmount, registrationFee))
+	if fetchErr != nil {
+		return log, nil, fetchErr
+	}
+
+	owner := actCtx.Caller
+	if act.OwnerAddress() != nil {
+		owner = act.OwnerAddress()
+	}
+
+	// cannot collide with existing operator
+	if _, ok := esm.getExecutor(ExecutorTypeProposer, act.OperatorAddress()); ok {
+		return log, nil, &handleError{
+			err:           ErrInvalidOperator,
+			failureStatus: iotextypes.ReceiptStatus_ErrCandidateConflict,
+		}
+	}
+
+	bucket := NewVoteBucket(act.OperatorAddress(), owner, actAmount, act.Duration(), blkCtx.BlockTimeStamp, act.AutoStake())
+	bucketIdx, err := esm.putBucket(bucket)
+	if err != nil {
+		return log, nil, err
+	}
+	log.AddTopics(byteutil.Uint64ToBytesBigEndian(bucketIdx), owner.Bytes())
+
+	c := &Executor{
+		Owner:     owner,
+		Operator:  act.OperatorAddress(),
+		Reward:    act.RewardAddress(),
+		Type:      ExecutorTypeProposer,
+		BucketIdx: bucketIdx,
+		Amount:    actAmount,
+	}
+
+	if err := esm.putExecutor(c); err != nil {
+		return log, nil, csmErrorToHandleError(owner.String(), err)
+	}
```

Comment:
you will need to do a `DebitBucketPool(actAmount, true)` call to add `actAmount` to bucket pool and balance the book (same as `handleCandidateRegister`), since we are taking in `actAmount` tokens